internal/buffer/settings.go

```diff
@@ -18,6 +18,8 @@ func (b *Buffer) ReloadSettings(reloadFiletype bool) {
 	if reloadFiletype && !local && !volatile {
 		// need to update filetype before updating other settings based on it
 		b.Settings["filetype"] = "unknown"
+		settings["filetype"] = "unknown"
```

Comment:
Is this line a leftover you forgot to remove? AFAICS it is buggy in several ways, and furthermore completely unneeded, since https://github.com/zyedidia/micro/pull/3662/commits/b40854a860e6abae69dc64e9eb61f4e59b7ed6c6? (Which btw means that https://github.com/zyedidia/micro/pull/3662/commits/b40854a860e6abae69dc64e9eb61f4e59b7ed6c6 should come before this commit, not after it.)

Also about the commit message: "buffer: `Fix InitLocalSettings()` on reload/changed `filetypes`" - AFAICS that is not what this commit actually does? Like, at all?